runtime/runtime/src/actions.rs

```diff
@@ -444,16 +441,10 @@ pub(crate) fn action_implicit_account_creation_transfer(
             * near_primitives::account::AccessKey::ACCESS_KEY_NONCE_RANGE_MULTIPLIER;
     }
 
-    // 0 for ED25519
-    let mut public_key_data = Vec::with_capacity(33);
-    public_key_data.push(0u8);
-    public_key_data.extend(
-        hex::decode(account_id.as_ref().as_bytes())
-            .expect("account id was a valid hex of length 64 resulting in 32 bytes"),
-    );
-    debug_assert_eq!(public_key_data.len(), 33);
-    let public_key = PublicKey::try_from_slice(&public_key_data)
-        .expect("we should be able to deserialize ED25519 public key");
+    // NOTE: The account_id is hex like, because we've checked the permissions before.
+    debug_assert!(account_id.is_implicit());
```

Comment:
This debug assert doesn't do anything meaningful at this point, given the unwrap below is going to execute the same check and panic regardless of debug-assertions being enabled or disabled.